internal/olm/operator/registry/index_image.go

```diff
@@ -102,36 +86,41 @@ func (c IndexImageCatalogCreator) getDBPath(ctx context.Context) (string, error)
 	return defaultDBPath, nil
 }
 
-func (c IndexImageCatalogCreator) createRegistryPod(ctx context.Context, dbPath string) (*index.RegistryPod, error) {
-	// Create registry pod, assigning its owner as the catalog source
-	registryPod, err := index.NewRegistryPod(c.cfg.Client, dbPath, c.BundleImage, c.cfg.Namespace)
+func (c IndexImageCatalogCreator) createRegistryPod(ctx context.Context, dbPath string, cs *v1alpha1.CatalogSource) (*corev1.Pod, error) {
+	// Initialize registry pod
+	registryPod, err := index.NewRegistryPod(c.cfg, dbPath, c.BundleImage, c.cfg.Namespace)
 	if err != nil {
-		return nil, fmt.Errorf("error in initializing registry pod")
+		return nil, fmt.Errorf("error in initializing registry pod: %v", err)
 	}
 
-	if err = registryPod.Create(ctx); err != nil {
-		return nil, fmt.Errorf("error in creating registry pod")
+	var pod *corev1.Pod
+	// Create registry pod
+	if pod, err = registryPod.Create(ctx, cs); err != nil {
+		return nil, fmt.Errorf("error in creating registry pod: %v", err)
 	}
 
-	return registryPod, nil
+	return pod, nil
 }
 
-func (c IndexImageCatalogCreator) updateCatalogSource(podAddr string, cs *v1alpha1.CatalogSource) error {
+func (c IndexImageCatalogCreator) updateCatalogSource(ctx context.Context, podAddr string, cs *v1alpha1.CatalogSource) error {
 	// Update catalog source with source type as grpc and address to point to the pod IP
-	cs.Spec.SourceType = v1alpha1.SourceTypeGrpc
-	cs.Spec.Address = index.GetRegistryPodHost(podAddr)
-
-	// Update catalog source with annotations for index image,
-	// injected bundle, and registry add mode
-	injectedBundlesJSON, err := json.Marshal(c.InjectBundles)
-	if err != nil {
-		return fmt.Errorf("error in json marshal injected bundles: %v", err)
+	catsrcKey := types.NamespacedName{
+		Namespace: cs.GetNamespace(),
+		Name:      cs.GetName(),
 	}
-	cs.ObjectMeta.Annotations = map[string]string{
-		"operators.operatorframework.io/index-image":        c.IndexImage,
-		"operators.operatorframework.io/inject-bundle-mode": c.InjectBundleMode,
-		"operators.operatorframework.io/injected-bundles":   string(injectedBundlesJSON),
```

Comment:
@joelanford these are no longer needed?

Reply:
I removed it because I think they are not needed for run bundle. Will be needed for run bundle upgrade, IIRC.

Reply:
We'll need `run bundle` to add these annotations so that `run bundle-upgrade` can use them.

We should go ahead and add them so that they'll already be present on existing catalog sources created with `run bundle` when we deliver the `run bundle-upgrade` command.

Reply:
Sounds good, thanks for confirming. I've added the annotations back.